### ISSUE TITLE
attention mask

### ISSUE THREAD

**Issue by jos-fisher-jejgkej** [Yxxxb/VoCo-LLaMA#24]
Hello, thank you for your work, it inspires me a lot. Regarding the attetion mask, I have a question, the attention mask of the pad token for all tokens seems to be set to true, since the left padding is set in the code (this part is different from llava, is it an additional setting?). Therefore, even if the pad part participates in the attention calculation, it will not inject information into the text token or visual token. I don't know if my understanding is correct, and if the attention in the pad token section can be removed to reduce the computational load. @Yxxxb 
https://github.com/Yxxxb/VoCo-LLaMA/blob/3b6e14d9fe9d6b5793c48aa2665aeca45660a082/llava/model/language_model/llava_llama_1stg.py#L297

**Comment by Yxxxb** [comment]
Thanks for your interest of our work!

This is a minor detail issue. We found that using a 4D attn mask in sdpa requires left padding. Our initial design was LLaVA's default right-padding settings, but they were not available when constructing the 4D attn mask.

Here is the specific code:
https://github.com/Yxxxb/VoCo-LLaMA/blob/main/llava/model/language_model/cache_py/modeling_attn_mask_utils.py#L189
Possible causes and solutions:
https://github.com/pytorch/pytorch/issues/110213

It has been a while since I wrote this part of the code, so I may not remember the exact line numbers, but I am certain that the problem occurred when constructing the 4D mask and performing the attn calculation. I clearly remember that I specifically modified the padding method for this purpose. If you have any further questions, please raise them in the issue.
